src/ui/test/qa_dialog.cpp

```diff
@@ -0,0 +1,84 @@
+#include "imgui.h"
+#include "imgui_test_engine/imgui_te_context.h"
+
+#include "components/Dialog.hpp"
+#include <boost/ut.hpp>
+
+#include "ImGuiTestApp.hpp"
+
+using namespace boost;
+using namespace DigitizerUi::components;
+
+class TestApp : public DigitizerUi::test::ImGuiTestApp {
+private:
+    using DigitizerUi::test::ImGuiTestApp::ImGuiTestApp;
+
+    void registerTests() override {
+        struct TestState {
+            bool         okEnabled = true;
+            DialogButton pressedButton;
+        };
+
+        ImGuiTest* t = IM_REGISTER_TEST(engine(), "dialog", "test1");
```

Comment:
Naming:
- Might be better for later to prefix component tests e.g. "components::dialog".
- "test1" -> "testButtonStates" ?